awx/main/tasks/receptor.py

```diff
@@ -641,29 +641,33 @@ def kube_config(self):
 
 @task()
 def write_receptor_config():
-    receptor_config = list(RECEPTOR_CONFIG_STARTER)
-
-    instances = Instance.objects.filter(node_type=Instance.Types.EXECUTION)
-    for instance in instances:
-        peer = {'tcp-peer': {'address': f'{instance.hostname}:{instance.listener_port}', 'tls': 'tlsclient'}}
-        receptor_config.append(peer)
-
     lock = FileLock(__RECEPTOR_CONF_LOCKFILE)
     with lock:
+        receptor_config = list(RECEPTOR_CONFIG_STARTER)
+
+        this_inst = Instance.objects.me()
+        instances = Instance.objects.filter(node_type=Instance.Types.EXECUTION).exclude(node_state=Instance.States.DEPROVISIONING)
+        for instance in instances:
+            peer = {'tcp-peer': {'address': f'{instance.hostname}:{instance.listener_port}', 'tls': 'tlsclient'}}
+            receptor_config.append(peer)
+
         with open(__RECEPTOR_CONF, 'w') as file:
             yaml.dump(receptor_config, file, default_flow_style=False)
 
-    receptor_ctl = get_receptor_ctl()
+        receptor_ctl = get_receptor_ctl()
 
-    attempts = 10
-    backoff = 1
-    for attempt in range(attempts):
-        try:
-            receptor_ctl.simple_command("reload")
-            break
-        except ValueError:
-            logger.warning(f"Unable to reload Receptor configuration. {attempts-attempt} attempts left.")
-            time.sleep(backoff)
-            backoff += 1
-    else:
-        raise RuntimeError("Receptor reload failed")
+        attempts = 10
+        backoff = 1
+        for attempt in range(attempts):
+            try:
+                receptor_ctl.simple_command("reload")
+                break
+            except ValueError:
+                logger.warning(f"Unable to reload Receptor configuration. {attempts-attempt} attempts left.")
+                time.sleep(backoff)
+                backoff += 1
+        else:
+            raise RuntimeError("Receptor reload failed")
+
+        InstanceLink.objects.filter(source=this_inst).exclude(target__in=instances).delete()
```

Comment:
Wait, why are you deleting these?

Reply:
That line of code was intended to deal with links to a node that was just deprovisioned (we are excluding those links to nodes that are still there).  However, I ultimately decided to just let Django's cascade delete deal with them.